.github/workflows/python-app.yml

```diff
@@ -26,32 +30,40 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
-
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install flake8 pytest mypy pylint pytest-cov
+        pip install flake8 pytest mypy pylint pytest-cov pytest-xdist
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
         pip install -e .[test]
 
     - name: Lint with flake8
+      if: ${{ env.TEST_ONLY_FAST != 'true' }}
       run: |
         # stop the build if there are Python syntax errors or undefined names
         flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics || echo "failed lint"
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics || echo "failed second lint"
 
     - name: PyLint - more lint
+      if: ${{ env.TEST_ONLY_FAST != 'true' }}
       run: |
         pylint -E cdci_data_analysis || echo 'this is fine!'
 
     - name: MyPy
+      if: ${{ env.TEST_ONLY_FAST != 'true' }}
       run: |
         mypy cdci_data_analysis --ignore-missing-imports || echo 'this is fine too!'
 
-    - name: Test with pytest
+    - name: Test fast subset with pytest
       run: |
-        python -m pytest tests --cov=cdci_data_analysis -sv --full-trace --log-cli-level=DEBUG -m 'not test_drupal'
+        python -m pytest tests --cov=cdci_data_analysis -sv --full-trace --log-cli-level=DEBUG -m "not test_drupal and fast" --durations=0
+
+    - name: Test everything with pytest
+      if: ${{ env.TEST_ONLY_FAST != 'true' }}
+      run: |
+        python -m pytest tests --cov=cdci_data_analysis -sv --full-trace --log-cli-level=DEBUG -m "not test_drupal" --durations=0
```

Comment:
perhaps I would also have the `fast` tests not run since they are run in the previous step

Reply:
I left them since there might be some unwanted interaction.

Also, in fast mode, all dispatchers are long-living.

https://github.com/oda-hub/dispatcher-app/pull/318/files#diff-87a0b6a348b0b79bdf41f844da34925be0474bf965cc3a09ff199e93f0628d93R582-R584

So it helps to re-run them in normal mode, in case there is some unwanted state preservation.